integration/container/devices_windows_test.go

```diff
@@ -0,0 +1,158 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/integration/internal/container"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/poll"
+	"gotest.tools/v3/skip"
+)
+
+// TestWindowsDevices that Windows Devices are correctly propagated
+// via HostConfig.Devices through to the implementation in hcsshim.
+func TestWindowsDevices(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "windows")
+	defer setupTest(t)()
+	client := testEnv.APIClient()
+	ctx := context.Background()
+
+	testData := []struct {
+		doc                         string
+		devices                     []string
+		isolation                   containertypes.Isolation
+		expectedStartFailure        bool
+		expectedStartFailureMessage string
+		expectedExitCode            int
+		expectedStdout              string
+		expectedStderr              string
+	}{
+		{
+			"process/no device mounted",
+			[]string{},
+			containertypes.IsolationProcess,
+			false,
+			"",
+			1,
+			"",
+			"",
```

Comment:
I generally prefer being explicit for these, and to include the keys for each fields (it makes it easier to see what field is what, and doesn't require fields to be set to an empty value (if it's the default)